libmodernize/main.py

```diff
@@ -17,6 +18,35 @@
 from libmodernize import __version__
 from libmodernize.fixes import lib2to3_fix_names, six_fix_names, opt_in_fix_names
 
+
+class LFPreservingRefactoringTool(StdoutRefactoringTool):
+    """ https://github.com/python-modernize/python-modernize/issues/121 """
+    def write_file(self, new_text, filename, old_text, encoding):
+        # detect linefeeds
+        lineends = {'\n':0, '\r\n':0, '\r':0}
+        lines = []
+        for line in open(filename, 'rb'):
+            if line.endswith('\r\n'):
+                lineends['\r\n'] += 1
+            elif line.endswith('\n'):
+                lineends['\n'] += 1
+            elif line.endswith('\r'):
+                lineends['\r'] += 1
+            lines.append(line.rstrip('\r\n'))
+        super(LFPreservingRefactoringTool, self).write_file(
+            new_text, filename, old_text, encoding)
+        # detect if line ends are consistent in source file
+        if sum([bool(lineends[x]) for x in lineends]) == 1:
+            # detect if line ends are different from system-specific
+            newline = [x for x in lineends if lineends[x] != 0][0]
+            if os.linesep != newline:
+                with open(filename, 'wb') as f:
+                    for line in lines:
```

Comment:
You're using the lines that you read *before* the file was rewritten, so this will undo the changes modernize made. You need to base it on `new_text`, or re-read the file after it is changed.